src/open_r1/rewards.py

```diff
@@ -367,6 +368,12 @@ def extract_code(completion: str, language: str = "python") -> str:
     return extracted_answer
 
 
+def binary_code_reward(completions, **kwargs) -> list[float]:
```

Comment:
Should we add a slow unit test for this?

Reply:
This would require code examples which pass only some of the unit tests. I could probably find these in some of our synthetic datasets, but it would take a while and I need to focus on the async exception refactor.